sklearn/metrics/classification.py

```diff
@@ -313,10 +317,24 @@ class labels [2].
            computational linguistics". Computational Linguistic 34(4):555-596.
     """
     confusion = confusion_matrix(y1, y2, labels=labels)
-    P = confusion / float(confusion.sum())
-    p_observed = np.trace(P)
-    p_expected = np.dot(P.sum(axis=0), P.sum(axis=1))
-    return (p_observed - p_expected) / (1 - p_expected)
+    n = confusion.shape[0]
+    sum0 = np.sum(confusion, axis=0)
+    sum1 = np.sum(confusion, axis=1)
+    expected = np.outer(sum0, sum1) / np.sum(sum0)
+
+    if weights == None:
+        w_mat = np.ones([n, n]) - np.diag(np.ones([n, ]))
```

Comment:
oups it seems n is number of classes. That's why it should have been called n_classes...